digits/extensions/data/interface.py

```diff
@@ -109,3 +109,27 @@ def itemize_entries(self, stage):
         this function in no particular order
         """
         raise NotImplementedError
+
+    @staticmethod
+    def can_explore():
```

Comment:
why don't you have `get_data()` return `None` by default and check the return value on caller side? This way, sub-classes only have to override `get_data()`